client/src/elm/Pages/Prenatal/Utils.elm

```diff
@@ -2543,60 +2643,404 @@ outsideCareDiagnosesWithPossibleMedication =
+                , referralFormHospital = or form.referralFormHospital (resolveFromFacilitySignsValue ReferralFormHospital)
+                , referToMentalHealthSpecialist = or form.referToMentalHealthSpecialist (resolveFromFacilitySignsValue ReferToMentalHealthSpecialist)
+                , referralFormMentalHealthSpecialist = or form.referralFormMentalHealthSpecialist (resolveFromFacilitySignsValue ReferralFormMentalHealthSpecialist)
+                , accompanyToMentalHealthSpecialist = or form.accompanyToMentalHealthSpecialist (resolveFromFacilitySignsValue AccompanyToMentalHealthSpecialist)
+                , referToARVProgram = or form.referToARVProgram (resolveFromFacilitySignsValue ReferToARVProgram)
+                , referralFormARVProgram = or form.referralFormARVProgram (resolveFromFacilitySignsValue ReferralFormARVProgram)
+                , accompanyToARVProgram = or form.accompanyToARVProgram (resolveFromFacilitySignsValue AccompanyToARVProgram)
+                , referToNCDProgram = or form.referToNCDProgram (resolveFromFacilitySignsValue ReferToNCDProgram)
+                , referralFormNCDProgram = or form.referralFormNCDProgram (resolveFromFacilitySignsValue ReferralFormNCDProgram)
+                , accompanyToNCDProgram = or form.accompanyToNCDProgram (resolveFromFacilitySignsValue AccompanyToNCDProgram)
+                , facilityNonReferralReasons = or form.facilityNonReferralReasons value.facilityNonReferralReasons
                 }
             )
 
 
-toPrenatalSendToHCValueWithDefault : Maybe PrenatalSendToHCValue -> Maybe ReferralFacility -> SendToHCForm -> Maybe PrenatalSendToHCValue
-toPrenatalSendToHCValueWithDefault saved referralFacility form =
-    prenatalSendToHCFormWithDefault form saved
-        |> toPrenatalSendToHCValue referralFacility
+toPrenatalReferralValueWithDefault : Maybe PrenatalReferralValue -> ReferralForm -> Maybe PrenatalReferralValue
+toPrenatalReferralValueWithDefault saved form =
+    referralFormWithDefault form saved
+        |> toPrenatalReferralValue
 
 
-toPrenatalSendToHCValue : Maybe ReferralFacility -> SendToHCForm -> Maybe PrenatalSendToHCValue
-toPrenatalSendToHCValue referralFacility form =
+toPrenatalReferralValue : ReferralForm -> Maybe PrenatalReferralValue
+toPrenatalReferralValue form =
     let
-        signs =
+        sendToHCSigns =
             [ ifNullableTrue HandReferrerForm form.handReferralForm
             , ifNullableTrue ReferToHealthCenter form.referToHealthCenter
             , ifNullableTrue PrenatalAccompanyToHC form.accompanyToHealthCenter
             ]
                 |> Maybe.Extra.combine
-                |> Maybe.map (List.foldl EverySet.union EverySet.empty >> ifEverySetEmpty NoSendToHCSigns)
+                |> Maybe.map (List.foldl EverySet.union EverySet.empty)
+
+        referToFacilitySigns =
+            [ ifNullableTrue ReferToHospital form.referToHospital
```

Comment:
@anvmn just asking here... Can `ifNullableTrue` be extracted from this list and applied with a `map`?

Reply:
@mariano-dagostino 
I suppose it is possible, but I'll have to see the code you're suggesting to give better answer.